src/errors/customError.ts

```diff
@@ -0,0 +1,11 @@
+export class CustomError extends Error {
+  code: number;
+  additionalInfo?: string;
+
+  constructor(code: number, message: string, additionalInfo?: string) {
```

Comment:
o argumento poderia ser um objeto com esses campos, dessa forma é possível quem for usar o erro confundir a ordem da mensagem e das informações adicionais